containers/EmpSponsors.ts

```diff
@@ -18,56 +22,56 @@ interface EmpMap {
   [address: string]: SponsorMap;
 }
 
+interface FinancialContractQuery {
+  id: string;
+  sponsorPositions: PositionState;
+}
+
 const useEmpSponsors = () => {
   const { block$ } = Connection.useContainer();
   const { contract: emp } = EmpContract.useContainer();
+  const { loading, error, data } = useQuery(ACTIVE_POSITIONS);
 
   const [activePositions, setActivePositions] = useState<EmpMap>({});
 
   // get position information about every sponsor that has ever created a position.
   const querySponsors = async () => {
     if (emp) {
-      const newSponsorFilter = emp.filters.NewSponsor(null);
-      const newSponsorEvents = await emp.queryFilter(newSponsorFilter);
-
-      // Create a clone of the active-positions mapping that we can update.
-      let newPositions = { ...activePositions };
-
-      // Map all active sponsor information to each EMP.
-      if (!newPositions[emp.address]) {
-        newPositions[emp.address] = {};
+      if (error) {
+        console.error(`Apollo client failed to fetch graph data:`, error);
       }
+      if (!loading && data) {
+        // Create a clone of the active-positions mapping that we can update.
+        let newPositions = { ...activePositions };
 
-      newSponsorEvents.forEach(async (e: Event) => {
-        const sponsorAddress = e.args?.sponsor;
-
-        if (!newPositions[emp.address][sponsorAddress]) {
-          // Check if sponsor has a current position. Current positions have locked
-          // a non-zero amount of collateral.
-          const res = await Promise.all([
-            emp.positions(sponsorAddress),
-            emp.getCollateral(sponsorAddress),
-          ]);
-          const positionData = res[0];
-          const collateral = res[1];
-
-          // Update the active positions mapping.
-          if (collateral[0].gt(0)) {
-            newPositions[emp.address][sponsorAddress] = {
-              tokensOutstanding: positionData.tokensOutstanding[0],
-              lockedCollateral: collateral[0],
+        // Map all active sponsor information to each EMP.
+        if (!newPositions[emp.address]) {
+          newPositions[emp.address] = {};
+        }
+
+        data.financialContracts
+          .find(
+            (contract: FinancialContractQuery) =>
+              utils.getAddress(contract.id) === emp.address
+          )
+          .sponsorPositions.forEach((position: PositionState) => {
+            const address = utils.getAddress(position.id.split("-")[0]);
```

Comment:
What is the format of the position `id` field?

Reply:
The "id" is a string that concatenates the sponsor's address with the financial-contrct's address. We could also grab the sponsor { id } from graph ql instead of just asking for id.

But this made defining the PositionState interface slightly easier (one fewer level)